### PR TITLE
Fix encryption/batching bug

### LINKED ISSUE #4170: Flaky test - TestMessengerCommunitiesTokenPermissionsSuite/TestBecomeMemberPermissions
```
2023-10-19T13:26:03.152Z	DEBUG	node/node.go:185	Epoch processing	{"mvds": {"node": "04a3eebe", "epoch": 480}}
    communities_messenger_token_permissions_test.go:715: 
        	Error Trace:	/home/jenkins/workspace/status-go_prs_tests_PR-4163/protocol/communities_messenger_token_permissions_test.go:715
        	Error:      	Received unexpected error:
        	            	no messages
        	Test:       	TestMessengerCommunitiesTokenPermissionsSuite/TestBecomeMemberPermissions
```

### PR DESCRIPTION
**Pull request by cammellos**
Fix issues with out of order encryption and batched messages
This commit fixes 3 issues:

1) In some cases, the hash ratchet was not correctly found
2) Out of order messages were not processed correctly as the wrong error
   was returned
3) Batched non datasync messages were not processed correctly

Fixes: https://github.com/status-im/status-go/issues/4170

### CODE REVIEW

File: protocol/common/message_sender.go [hunk 1]
```diff
@@ -782,11 +782,16 @@ func (s *MessageSender) HandleMessages(shhMessage *types.Message) ([]*v1protocol
...
+			err = s.handleEncryptionLayer(context.Background(), &statusMessage)
+			if err != nil {
+				hlogger.Debug("failed to handle an encryption message", zap.Error(err))
+			}
+
 			stms, as, err := unwrapDatasyncMessage(&statusMessage, s.datasync)
 			if err != nil {
 				hlogger.Debug("failed to handle datasync message", zap.Error(err))
 				//that wasn't a datasync message, so use the original payload
-				statusMessages = append(stms, &statusMessage)
```
Comment (vitvly) [inline protocol/common/message_sender.go:789]: Ah, so this is the wrong loop variable being used?

File: protocol/common/raw_messages_persistence.go [hunk 2]
```diff
@@ -292,14 +292,14 @@ func (db RawMessagesPersistence) InsertPendingConfirmation(confirmation *RawMess
 }
 
 func (db RawMessagesPersistence) SaveHashRatchetMessage(groupID []byte, keyID []byte, m *types.Message) error {
-	_, err := db.db.Exec(`INSERT INTO hash_ratchet_encrypted_messages(hash, sig, TTL, timestamp, topic, payload, dst, p2p, group_id, key_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`, m.Hash, m.Sig, m.TTL, m.Timestamp, types.TopicTypeToByteArray(m.Topic), m.Payload, m.Dst, m.P2P, groupID, keyID)
+	_, err := db.db.Exec(`INSERT INTO hash_ratchet_encrypted_messages(hash, sig, TTL, timestamp, topic, payload, dst, p2p, padding, group_id, key_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)`, m.Hash, m.Sig, m.TTL, m.Timestamp, types.TopicTypeToByteArray(m.Topic), m.Payload, m.Dst, m.P2P, m.Padding, groupID, keyID)
```
Comment (vitvly) [inline protocol/common/raw_messages_persistence.go:295]: Sorry, didn't quite understand where `padding` is used.

**Review by vitvly — APPROVED** [review]

File: protocol/common/message_sender.go (hunk 1 above)
Comment (cammellos) [reply protocol/common/message_sender.go:789]: that's right, it was using the wrong variable

File: protocol/common/raw_messages_persistence.go (hunk 2 above)
Comment (cammellos) [reply protocol/common/raw_messages_persistence.go:295]: it's a field in the envelope, it's needed for properly decrypting it

**Review by jrainville — COMMENTED** [review]
Great job!

Hopefully, this both fixes most flaky tests + it also improves the message reliability!

**Pull request merged**